### PR TITLE
refactor: update GROQ query for document versions

### PR DESCRIPTION
**Pull request by RitaDias**
### Description

- Remove dummy getter
- Set up a listener for the new groq query (warning: it's really slow)
- Update usage on DocumentVersion and Footer actions
- Update tests
- Add hook test
- Remove dummyGetters & separate methods
- Rename "dummy getters" rip

Heads up: hook test is bare bones, I didn't have time to do more (maintenance work)

### CODE REVIEW

File: packages/sanity/src/core/bundles/components/dialog/CreateBundleDialog.tsx
Comment (RitaDias) [inline packages/sanity/src/core/bundles/components/dialog/CreateBundleDialog.tsx:38]: if the versioned item of this document already exists then we shouldn't add it again

File: packages/sanity/src/core/bundles/components/dialog/CreateBundleDialog.tsx
Comment (jordanl17) [inline packages/sanity/src/core/bundles/components/dialog/CreateBundleDialog.tsx:1]: Sorry again @RitaDias  - I think these changes will need to be made to `BundleDetailsDialog` instead of here

Comment (RitaDias) [reply packages/sanity/src/core/bundles/components/dialog/CreateBundleDialog.tsx:1]: No need to apologise, I'll do it! :)

File: packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx
Comment (bjoerge) [inline packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:42]: We tend to call this "publishedId" elsewhere

File: packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx
Comment (bjoerge) [inline packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:43]: There's already a `getPublishedId()` in this package, could we reuse that instead?

**Review by bjoerge — COMMENTED** [review]
Solid work! Just a couple of comments/suggestions, nothing big. Thanks for adding tests as well!

File: packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx
Comment (RitaDias) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:43]: getPublishedId only accounts for the drafts.(id) structure.
Would we be ok with changing this? I just thought about this way because I was concerned about changing those things 🤔

Comment (bjoerge) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:43]: That's a good point. Could we add a new function (`getBundlePublishedId()`?) that includes the logic for bundle ids for now? Id like for us to have a central place to keep this logic instead of spreading it around.

Comment (RitaDias) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:43]: Ahhh I actually see that @juice49  added something for this already in the getPublishedId
image

I'm thinking about moving this to a `getbundlePublishedId`, thoughts?

Comment (bjoerge) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx]: Seems to have caused a test failure 🤔

Comment (RitaDias) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx]: Just noticed, forgot to add the mock!
Should be fine now (just pushed the update)

File: packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx
Comment (pedrobonamin) [inline packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:35]: could this be replaced by the `useListener` hook? It seems to handle the same logic.

Comment (pedrobonamin) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:35]: Or for `listenQuery` ? And maybe we could update the projection to get only the id of the documents?

Comment (pedrobonamin) [inline packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:35]: Have you considered here instead of setting the listener, use the `listenQuery` function? 
Also, probably we can reduce the data returned from the query, projecting it to include only the id
 `*[sanity::versionOf("${publishedId}")]{_id}`

Comment (pedrobonamin) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:35]: Sorry if I'm missing something, but this looks like it could be handled with the `listenQuery` observer and simplify this hook. 

```
export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPerspectiveState {
  const {documentId} = props

  const client = useClient(DEFAULT_STUDIO_CLIENT_OPTIONS)
  const {data: bundles} = useBundles()
  // TODO: How to identify documents with multiple dots ids, maybe it will be better to provide the whole document?
  const publishedId = getPublishedId(documentId, documentId.includes('.'))

  const QUERY = `*[sanity::versionOf("${publishedId}")]{_id, _version}`

  const observable = useMemo(
    () => listenQuery(client, QUERY) as Observable<Pick<SanityDocument, '_id' | '_version'>[]>,
    [QUERY, client],
  )

   // This will return the sanity documents that are a version of the published document.
  const versions = useObservable(observable, null)

  const state = versions
    ? versions
        .map((r) => bundles?.find((b) => r._version && getBundleSlug(r._id) === b.slug))
        .filter((b) => b !== undefined)
    : null

  return {
    data: state,
  }
}

```

Comment (bjoerge) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:35]: I thought about this too, but I don't think it's important enough to block merging, and that it's something that can be addressed down the line? I think it makes sense for us to do a separate refactor/optimization pass to unify some of the data fetching patterns at a later point.

File: packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx
Comment (bjoerge) [reply packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx:25]: Good point. I'll leave it to @RitaDias to decide whether to do that in a follow-up PR or before merging in this one.

File: packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx
Comment (jordanl17) [inline packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx:52]: `BundleMenu` seems to accepts `BundleDocument[] | null`, and I think that's also the type of `documentVersions`, so might not need to do the type assertion here

File: packages/sanity/src/core/bundles/hooks/__tests__/useDocumentVersions.test.tsx
Comment (jordanl17) [inline packages/sanity/src/core/bundles/hooks/__tests__/useDocumentVersions.test.tsx:1]: Nice tests for this hook!!

Comment (RitaDias) [reply packages/sanity/src/core/bundles/hooks/__tests__/useDocumentVersions.test.tsx:1]: ❤️

File: packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx
Comment (RitaDias) [reply packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx:52]: You're right, might have been because of something I did before messing around with the types, removed it :)

File: packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx
Comment (pedrobonamin) [reply packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx:25]: re `BundleActions` I think they will eventually change, specially after the demo, probably we will want it in a more obvious place, but that won't be decided until Marius is back.
Not sure how that change will impact this decision, but I think moving it to the PaneProvider will make it available for all the possible consumers. It can be done in a follow-up PR once you are back from support

Comment (RitaDias) [reply packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx:25]: WIll create a Linear story since I'll be on holiday next week 👍

File: packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx
Comment (pedrobonamin) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:35]: Agree, this suggestion is non blocking, it can be addressed later.

File: packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx
Comment (RitaDias) [reply packages/sanity/src/structure/panes/document/documentPanel/header/perspective/DocumentPerspectiveMenu.tsx:25]: Linear story created 👍

File: packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx
Comment (RitaDias) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:31]: I've updated with the suggestion from Pedro on slack :) 
Thanks once more!

Comment (RitaDias) [reply packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx:35]: Spoken on slack but had to drop the update due to time limitations (specifically around tests) created a linear story for it.

**Review by pedrobonamin — APPROVED** [review]
Great work @RitaDias , thanks!

**Pull request merged**
